browser/ui/views/frame/brave_system_menu_model_builder.cc

```diff
@@ -0,0 +1,36 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/views/frame/brave_system_menu_model_builder.h"
+
+#include "brave/app/brave_command_ids.h"
+#include "brave/grit/brave_generated_resources.h"
+#include "chrome/app/chrome_command_ids.h"
+#include "ui/base/models/simple_menu_model.h"
+
+BraveSystemMenuModelBuilder::BraveSystemMenuModelBuilder(
+    ui::AcceleratorProvider* provider,
+    Browser* browser)
+    : SystemMenuModelBuilder(provider, browser) {}
+
+BraveSystemMenuModelBuilder::~BraveSystemMenuModelBuilder() = default;
+
+void BraveSystemMenuModelBuilder::InsertBraveSystemMenuForBrowserWindow(
+    ui::SimpleMenuModel* model) {
+  std::optional<size_t> task_manager_index =
```

Comment:
need to check whether vertical tab is available or not for `browser`. If not, we shoud not have this menu entry.